src/components/Modals/EditExpertProfileModal.tsx

```diff
@@ -0,0 +1,166 @@
+
+  const [Avatar, setAvatar] = useState(props.avatar);
+
+  const avatarUploader = useRef(null);
+
+  function closeModal() {
+    modalContext.setOpenedModal(null);
+    // can be optional
+    [
+      setEmail,
+      setExpertise,
+      setDescription,
+      setExpertiseTags,
+      setVisibility,
+    ].map((method) => typeof method === 'function' && method(null));
+  }
+
+  const putExpertProfile = () => {
+    if (
+      email == null ||
+      expertise == null ||
+      description == null ||
+      expertiseTags == null ||
+      visibility == null
+    ) {
+      setFieldsMissing(true);
+      return;
+    }
+    apiContext
+      .updateJufoExpert(user.id, {
+        contactEmail: email,
+        description,
+        expertiseTags: expertiseTags.split(','),
+        active: visibility,
+      })
+      .then(() => closeModal());
+  };
+
+  return (
+    <DialogModalBase accentColor={accentColor}>
+      <DialogModalBase.Modal modalName="editExpertProfileModal">
+        <DialogModalBase.Header>
+          <div className={styles.avatarWrapper}>
+            <Avatar className={styles.avatar} />
+            <button
+              className={styles.avatarOverlay}
+              onClick={() => avatarUploader.current.click()}
+            >
+              <p>Ändern</p>
+              <input
+                type="file"
+                id="file"
+                ref={avatarUploader}
+                accept="image/*"
+                style={{ display: 'none' }}
+                onChange={(e) => {
+                  e.stopPropagation();
+                  e.preventDefault();
+                  const file = e.target.files[0];
+                  console.log(file);
+                  const fileReader = new FileReader();
+                  fileReader.onload = () => {
+                    setAvatar(() => ({ className }) => {
+                      return (
+                        <img
+                          className={className}
+                          alt="avatar"
+                          src={String(fileReader.result)}
+                        />
+                      );
+                    });
+                  };
+
+                  fileReader.readAsDataURL(file);
+                }}
+              />
+            </button>
+          </div>
+        </DialogModalBase.Header>
+        <DialogModalBase.Spacer />
+        <DialogModalBase.Header>
+          <DialogModalBase.Title>Melanie Meiers</DialogModalBase.Title>
+          <DialogModalBase.CloseButton />
+        </DialogModalBase.Header>
+        {fieldsMissing && (
+          <DialogModalBase.Error>
+            <span>Bitte alle Felder ausfüllen.</span>
+          </DialogModalBase.Error>
+        )}
+        <DialogModalBase.Form>
+          <DialogModalBase.InputCompound direction="vertical">
+            <DialogModalBase.TextBox
+              label="Kontakt E-Mail-Adresse"
+              onChange={(e) => setEmail(e.target.value)}
+              value={email}
```

Comment:
The email address field here should somehow have a very basic email address verification logic, e.g. check that at least a TLD and `@` is included. This way, we can reduce the amount of typos and therefore problems with the expert list. Many people use a comma instead of dot/full stop or accidentally omit the at-sign.

Reply:
Fixed in 3a687e1458e47e8e60698eadfb36d91abd70d3f1